### ISSUE TITLE
ARM64: Vector64<T> is converted to HVA TYP_DOUBLE

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#35976]
As we found out in https://github.com/dotnet/runtime/pull/35636#discussion_r421625643 , we are converting the return type of `Vectot64<T>` to HVA `TYP_DOUBLE`.  This impacts `Vector64.CreateScalar(uint)` or `byte`, `short`, etc when the result is used. 

```csharp
[MethodImpl(MethodImplOptions.NoInlining)]
public static uint fo1(uint value) {	
	var input = Vector64.CreateScalar(value);
	return AdvSimd.Extract(input, 0);
}
```

Generated code:

```asm
G_M6484_IG02:
        0F000410          movi    v16.2s, #0x00
        4E041C10          ins     v16.s[0], w0
        FD000FB0          str     d16, [fp,#24]
        FD400FB0          ldr     d16, [fp,#24]
        0E043E00          umov    w0, v16.s[0]
```
Below is the dump of where conversation happens:

```asm
impFixupStructReturnType: retyping
               [000039] ------------              *  HWINTRINSIC simd8  uint Insert
               [000038] ------------              \--*  LIST      void  
               [000034] ------------                 +--*  HWINTRINSIC simd8  uint get_Zero
               [000037] ------------                 \--*  LIST      void  
               [000035] ------------                    +--*  CNS_INT   int    0
               [000036] ------------                    \--*  LIST      void  
               [000009] ------------                       \--*  LCL_VAR   int    V00 arg0         

impFixupStructReturnType: result of retyping is
               [000039] ------------              *  HWINTRINSIC double uint Insert
               [000038] ------------              \--*  LIST      void  
               [000034] ------------                 +--*  HWINTRINSIC simd8  uint get_Zero
               [000037] ------------                 \--*  LIST      void  
               [000035] ------------                    +--*  CNS_INT   int    0
               [000036] ------------                    \--*  LIST      void  
               [000009] ------------                       \--*  LCL_VAR   int    V00 arg0         


               [000041] -A----------              *  ASG       double
               [000040] D------N----              +--*  LCL_VAR   double V05 tmp2         
               [000039] ------------              \--*  HWINTRINSIC double uint Insert
               [000038] ------------                 \--*  LIST      void  
               [000034] ------------                    +--*  HWINTRINSIC simd8  uint get_Zero
               [000037] ------------                    \--*  LIST      void  
               [000035] ------------                       +--*  CNS_INT   int    0
               [000036] ------------                       \--*  LIST      void  
               [000009] ------------                          \--*  LCL_VAR   int    V00 arg0  
```

**Comment by CarolEidt** [comment]
This is a dup of #35144

**Comment by CarolEidt** [comment]
I've submitted PR #37499.
With those changes, here's the full code generation for the `fo1` method above:
```
G_M6398_IG01:
        A9BF7BFD          stp     fp, lr, [sp,#-16]!
        910003FD          mov     fp, sp
                                                ;; bbWeight=1    PerfScore 1.50
G_M6398_IG02:
        0F000410          movi    v16.2s, #0x00
        4E041C10          ins     v16.s[0], w0
        0E043E00          umov    w0, v16.s[0]
                                                ;; bbWeight=1    PerfScore 2.50
G_M6398_IG03:
        A8C17BFD          ldp     fp, lr, [sp],#16
        D65F03C0          ret     lr
```

**Comment by CarolEidt** [comment]
Fixed with #37499
